source/common/common/assert.h

```diff
@@ -156,10 +162,20 @@ void resetEnvoyBugCountersForTest();
 // _ASSERT_VERBOSE, and this will call _ASSERT_VERBOSE,(__VA_ARGS__)
 #define ASSERT(...)                                                                                \
   EXPAND(_ASSERT_SELECTOR(__VA_ARGS__, _ASSERT_VERBOSE, _ASSERT_ORIGINAL)(__VA_ARGS__))
+
+#if !defined(NDEBUG) || defined(ENVOY_LOG_DEBUG_ASSERT_IN_RELEASE)
+// debug build or all ASSERTs compiled in release.
+#define SLOW_ASSERT(...) ASSERT(__VA_ARGS__)
+#else
+// Non-implementation of SLOW_ASSERTs when building only ENVOY_LOG_FAST_DEBUG_ASSERT_IN_RELEASE.
```

Comment:
nit: this path is also used for release builds, yes?

Reply:
no, only line 174-178 are for release builds.

This is for ENVOY_LOG_FAST_DEBUG_ASSERT_IN_RELEASE only. I needed to have ASSERT defined in this mode, so basically like 127-174 is for any case where an ASSERT is defined.